SuggestedConnection.js

```diff
@@ -1,26 +1,28 @@
 import React, { Component } from 'react'
+import NavigationService from './NavigationService.js';
 import { StyleSheet, Text, View, Image, Button, TouchableHighlight } from 'react-native'
 import Icon from 'react-native-vector-icons/Feather'
 
 export default class SuggestedConnection extends Component {
   
-  
   render() {
+    const user = this.props.suggestion
+    const { id, name, job_title, city } = this.props.suggestion
     return (
       <View style={styles.container}>
       <View style={styles.imageContainer}>
         <Image source={require('./profile-pic.jpeg')} style={styles.picture} />
       </View>
-        <TouchableHighlight onPress={() => this.props.viewProfile(this.props.suggestion.id)}>
-          <Text style={ styles.name }>{ this.props.suggestion.name }</Text>
+        <TouchableHighlight onPress={() => this.props.viewProfile(id)}>
+          <Text style={ styles.name }>{ name }</Text>
         </TouchableHighlight>
-        <Text style={ styles.position }>{ this.props.suggestion.job_title }</Text>
+        <Text style={ styles.position }>{ job_title }</Text>
         <View style={ styles.iconContainer }>
           <Icon name='map-pin' size={17} color='#93548F' style={ {marginRight: 2} } />
-          <Text style={ {color: 'white', marginLeft: 3} }>{ this.props.suggestion.city }</Text>
+          <Text style={ {color: 'white', marginLeft: 3} }>{ city }</Text>
         </View>
         <TouchableHighlight style={ styles.connectBtn }>
-          <Button title='Connect' color='white' />
+          <Button title='Connect' color='white' onPress={() => NavigationService.navigate('Modal', {user})} />
```

Comment:
Good find on the Navigation service 😄  I'm sure we can implement this in other places too!